--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,7 +53,7 @@
 
 # Set default C++ flags.
 include(CheckCCompilerFlag)
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -g -O3")
+#set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -g -O3")
 CHECK_C_COMPILER_FLAG("-std=gnu11" COMPILER_SUPPORTS_GNU11)
 if(CMAKE_C_STANDARD_COMPUTED_DEFAULT EQUAL "90")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99")
@@ -103,10 +103,10 @@
 option(BUILD_SHARED_LIBS
     "Build shared library. Set to OFF for static library." ON)
 # Unittest should be on for dev builds and off for releases.
-if(CMAKE_BUILD_TYPE MATCHES "Release")
+if(CMAKE_BUILD_TYPE MATCHES "Debug")
+    option(UNITTEST "Build unittest binaries." ON)
+else()
     option(UNITTEST "Build unittest binaries." OFF)
-else()
-    option(UNITTEST "Build unittest binaries." ON)
 endif()
 option(INSTALL_EXAMPLES "Install example code." OFF)
 if(INSTALL_EXAMPLES)
