From aac1858716caaef94488222bc72d9b3bf9061650 Mon Sep 17 00:00:00 2001
From: Thomas Schneider <qsx@chaotikum.eu>
Date: Sun, 22 Dec 2019 15:52:38 +0100
Subject: [PATCH 1/2] dracut.sh: Include kernel version in UEFI image metadata

Signed-off-by: Thomas Schneider <qsx@chaotikum.eu>
---
 dracut.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/dracut.sh b/dracut.sh
index 06c751dc..302ab9bf 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -1986,6 +1986,10 @@ if [[ $uefi = yes ]]; then
     [[ -s $dracutsysrootdir/usr/lib/os-release ]] && uefi_osrelease="$dracutsysrootdir/usr/lib/os-release"
     [[ -s $dracutsysrootdir/etc/os-release ]] && uefi_osrelease="$dracutsysrootdir/etc/os-release"
 
+    grep -v '^VERSION_ID' "$uefi_osrelease" > "$uefi_outdir/os-release.txt"
+    printf 'VERSION_ID="%s"\n' "$kernel" >> "$uefi_outdir/os-release.txt"
+    uefi_osrelease="$uefi_outdir/os-release.txt"
+
     if objcopy \
            ${uefi_osrelease:+--add-section .osrel=$uefi_osrelease --change-section-vma .osrel=0x20000} \
            --add-section .cmdline="${uefi_outdir}/cmdline.txt" --change-section-vma .cmdline=0x30000 \
-- 
2.24.1

