diff --git a/meson.build b/meson.build
index c9c11d5..86ff27d 100644
--- a/meson.build
+++ b/meson.build
@@ -34,7 +34,7 @@ else
 endif
 
 
-posixsignalmanager_lib = both_libraries('posixsignalmanager', 'PosixSignalManager.cpp',
+posixsignalmanager_lib = library('posixsignalmanager', 'PosixSignalManager.cpp',
                                         qt5.preprocess(moc_headers: 'PosixSignalManager.h'),
                                         dependencies : [qt5_dep, librt], cpp_args: macros,
                                         link_args : main_ld_vscript,
